SlowCheetah.JDT/Processors/JdtRecurse.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Sayed Ibrahim Hashimi. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See  License.md file in the project root for full license information.
+
+namespace SlowCheetah.JDT
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Newtonsoft.Json.Linq;
+
+    /// <summary>
+    /// Represents a recursive JDT transformation
+    /// </summary>
+    internal class JdtRecurse : JdtProcessor
+    {
+        /// <inheritdoc/>
+        public override string Verb { get; } = null;
+
+        /// <inheritdoc/>
+        public override void Process(JObject source, JObject transform)
+        {
+            // Nodes that should be removed from the transform after they are handled
+            List<string> nodesToRemove = new List<string>();
+
+            foreach (JProperty transformNode in transform.Properties()
+                .Where(p => p.Value.Type == JTokenType.Object && !JsonUtilities.IsJdtSyntax(p.Name)))
+            {
+                JToken sourceChild;
+                if (source.TryGetValue(transformNode.Name, out sourceChild) && sourceChild.Type == JTokenType.Object)
+                {
+                    ProcessTransform((JObject)sourceChild, (JObject)transformNode.Value);
+
+                    // If we have already recursed into that node, it should be removed from the transform
+                    nodesToRemove.Add(transformNode.Name);
+                }
+            }
+
+            // Remove all of the previously handled nodes
+            nodesToRemove.ForEach(node => transform.Remove(node));
```

Comment:
Because each transformation is containerized, later transformations need a way to know if a node has already been processed in case of a removal or rename in the source object. 
The transform object is not actually mutated as a deep clone is passed in, per [this](https://github.com/davilimap/slow-cheetah/pull/6/files#diff-5d0dbb0e3b7e365048b4495556f01a22R45).